### PR TITLE
Move tests that specify a target group to their own top-level directory.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Today when a test that specifies a target group is built it will place the
output for that target group in a subdirectory of the build with an empty
target group; this leads to a race condition in the build when
ArchiveTests is set to true.  Instead, place each test build in its own
directory; and in cases where TargetGroup is empty use netstandard as the
subdirectory name.

### CODE REVIEW

**Comment by chcosta** [comment]
LGTM

**Comment by jhendrixMSFT** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test please

**Comment by jhendrixMSFT** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test please

File: dir.props [hunk 1]
```diff
@@ -504,15 +504,15 @@
   <!-- Set up the default output and intermediate paths -->
   <PropertyGroup>
     <OSPlatformConfig>$(OSGroup).$(Platform).$(ConfigurationGroup)</OSPlatformConfig>
-    <TargetOutputRelPath Condition="'$(TargetGroup)'!=''">$(TargetGroup)/</TargetOutputRelPath>
+    <TargetOutputExtension Condition="'$(TargetGroup)'!=''">.$(TargetGroup)</TargetOutputExtension>
```
Comment (weshaggard) [inline dir.props]: As we chatted lets keep these in a sub-dir but for TargetGroup empty use netstandard as the folder name.

**Comment by ellismg** [comment]
@jhendrixMSFT The layout of the directories under the tests folder has changes with this (there's an intermediate netstandard folder now).  If that's expected, you'll need to pick up https://github.com/ellismg/corefx/commit/d977c50138c258f2a985036c0a0048342289eb20 as well.

**Comment by ellismg** [comment]
(once you get the run-test changes, the CentOS legs should go green).

**Comment by jhendrixMSFT** [comment]
@ellismg thanks!  I've picked up the change, rebased and resubmitted.

**Comment by jhendrixMSFT** [comment]
Merging to get the official build unblocked.

**Comment by weshaggard** [comment]
LGTM

**Comment by ericstj** [comment]
Why `netstandard` and not `dummysubdirectory`or `default` or `hackuponhackuponhack`.  As far as I can tell the directory has no meaning and you used a word that has a very specific meaning.  Not only does this not make any sense but it dilutes the meaning of the term and makes all the docs about test directory wrong.

>  this leads to a race condition in the build when ArchiveTests is set to true

Seems to me the bug here is that you are building a project with different global properties and same output directory.  Without understanding more about the race you are fixing that's what stands out as the problem.

**Pull request merged**